Bluepill-runner/Bluepill-runner/BPRunner.m

```diff
@@ -81,6 +86,150 @@ + (instancetype)BPRunnerWithConfig:(BPConfiguration *)config
+    } else {
+        // This is for testing in command line when we pass the xctestrun file
+        NSMutableSet *hostBundles = [[NSMutableSet alloc] init];
+        for (BPXCTestFile* bundle in testBundles) {
+            [hostBundles addObject:bundle.testHostPath];
+        }
+        if ([testBundles count] == 0) {
+            [BPUtils printInfo:ERROR withString:@"No host bundle founnd!"];
+        }
+        for (NSString *appPath in hostBundles) {
+            NSError *error = nil;
+            simulatorUDIDString = [self installApplicationwithHost:appPath withError:error];
+            if (!simulatorUDIDString || !error) {
+                [BPUtils printInfo:ERROR withString:@"Created simulator template and innstall applicationn failed with error: %@", error];
+                return FALSE;
+            }
+            [BPUtils printInfo:INFO withString:@"Created sim template: %@ for app host: %@", simulatorUDIDString, appPath];
+            self.testHostForSimUDID[appPath] = simulatorUDIDString;
+        }
+    }
+    return TRUE;
+}
+
+- (NSString *)installApplicationwithHost:(NSString *)testHost withError:(NSError *)error {
+    SimServiceContext *sc = [SimServiceContext sharedServiceContextForDeveloperDir:self.config.xcodePath error:&error];
+    if (!sc) {
+        [BPUtils printInfo:ERROR withString:[NSString stringWithFormat:@"SimServiceContext failed: %@", [error localizedDescription]]];
+        return nil;
+    }
+    SimDeviceSet *deviceSet = [sc defaultDeviceSetWithError:&error];
+    if (!deviceSet) {
+        [BPUtils printInfo:ERROR withString:[NSString stringWithFormat:@"SimDeviceSet failed: %@", [error localizedDescription]]];
+        return nil;
+    }
+    
+    self.simDeviceType = nil;
+    for (SimDeviceType *type in [sc supportedDeviceTypes]) {
+        if ([[type name] isEqualToString:@BP_DEFAULT_DEVICE_TYPE]) {
+            self.simDeviceType = type;
+            break;
+        }
+    }
+    
+    if (!self.simDeviceType) {
+        [BPUtils printInfo:ERROR withString:@"Invalid Device Type!"];
+        return nil;
+    }
+    
+    self.simRuntime = nil;
+    for (SimRuntime *runtime in [sc supportedRuntimes]) {
+        if ([[runtime name] isEqualToString:self.config.runtime]) {
+            self.simRuntime = runtime;
+            break;
+        }
+    }
+    
+    if (!self.simRuntime) {
+        [BPUtils printInfo:ERROR withString:@"Invalid runtime Type!"];
+        return nil;
+    }
+    
+    SimDevice *simDevice = [deviceSet createDeviceWithType:self.simDeviceType
+                                                   runtime:self.simRuntime
+                                                      name:@"BP_Simultator_Blueprint"
+                                                     error:&error];
+    [self.simDeviceTemplates addObject:simDevice];
+    if (!simDevice || error) {
+        [BPUtils printInfo:ERROR withString:@"Create simulator failed with error: %@", error];
+    } else {
```

Comment:
I think you want to `return nil` in the `if` here and move the entire `else` up an indent level.

Reply:
done